File: backend/app/schemas/inquiry.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class InquiryOut(BaseModel):
    """T3.11.23 — a chat, in the shape the client already reads.

    `id` is the chat's id now; `trip_id` is the trip the caller asked about, and
    it is **echoed from the request** rather than owned by the thread. That is
    the whole model change in one field: the trip used to be what the thread
    *was*, and is now what a message in it is about.

    `deal_id` is the open deal in this chat, if there is one. Several deals can
    live in one chat (owner's clarification 2026-09-07), so this is «the one to
    carry on in», not «the deal of this thread».
    """

    id: uuid.UUID
    trip_id: uuid.UUID | None = None
    sender_id: uuid.UUID
    carrier_id: uuid.UUID
    deal_id: uuid.UUID | None = None
    created_at: datetime
    model_config = ConfigDict(from_attributes=True)


class InquiryMessageCreate(BaseModel):
    text: str
    # T3.11.23 — which trip this message is about, when it is about one.
    #
    # The trip stopped being the thread's identity and became the content of the
    # message that raises it. Without this the column exists and nothing ever
    # fills it — and the carrier's panel, which counts «сколько человек
    # спросили про этот рейс», has nothing left to count.
    #
    # Optional because most messages are not about a trip: they are the rest of
    # the conversation.
    about_trip_id: uuid.UUID | None = None


class InquiryMessageOut(BaseModel):
    id: uuid.UUID
    inquiry_id: uuid.UUID
    sender_id: uuid.UUID
    text: str | None
    created_at: datetime
    model_config = ConfigDict(from_attributes=True)

File: frontend/src/api/inquiry.ts
import api from './client'
import type { Page } from './pagination'

export interface Inquiry {
  id: string
  trip_id: string
  sender_id: string
  carrier_id: string
  deal_id: string | null
  created_at: string
}

export interface InquiryMessage {
  id: string
  inquiry_id: string
  sender_id: string
  text: string | null
  created_at: string
}

export const openInquiry = (tripId: string) =>
  api.post<Inquiry>(`/api/trips/${tripId}/inquiry`)

export const listMyInquiries = () => api.get<Inquiry[]>('/api/inquiries')

export const listInquiryMessages = (
  inquiryId: string,
  params?: { after?: string; limit?: number },
) =>
  api.get<Page<InquiryMessage>>(`/api/inquiries/${inquiryId}/messages`, {
    params,
  })

/** T3.11.23 — `aboutTripId` is what keeps «пишу про этот рейс» expressible now
 *  that the thread belongs to the person rather than to the trip. Optional,
 *  because most messages are not about a trip: they are the rest of the
 *  conversation. */
export const postInquiryMessage = (
  inquiryId: string,
  text: string,
  aboutTripId?: string,
) =>
  api.post<InquiryMessage>(`/api/inquiries/${inquiryId}/messages`, {
    text,
    about_trip_id: aboutTripId,
  })

export const shareAddressInInquiry = (inquiryId: string, addressId?: string) =>
  api.post<InquiryMessage>(
    `/api/inquiries/${inquiryId}/messages/share-address`,
    addressId ? { address_id: addressId } : {},
  )
